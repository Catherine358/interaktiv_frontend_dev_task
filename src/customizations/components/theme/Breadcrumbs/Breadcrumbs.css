div.ui.secondary.vertical.segment.breadcrumbs {
  background-color: white;
  padding-right: 10px;
  padding-left: 10px;
  flex: 0.7;
}

.ui.vertical.segment {
  padding-right: 10px;
  padding-left: 10px;
}

div.ui.breadcrumb a.section,
div.ui.breadcrumb a,
div.ui.breadcrumb .section.active {
  color: #676767;
}

div.ui.breadcrumb a:hover {
  color: #676767;
}

div.ui.secondary.vertical.segment.breadcrumbs.admin {
  flex: 0.3;
  color: white;
  background-color: #676767;
}
