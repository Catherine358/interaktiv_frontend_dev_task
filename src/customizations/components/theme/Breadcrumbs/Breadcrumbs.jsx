import { Breadcrumb, Container, Segment } from 'semantic-ui-react';
import { Icon } from '@plone/volto/components';
import rightKey from '@plone/volto/icons/right-key.svg';

import './Breadcrumbs.css';

const Breadcrumbs = () => (
  <div className="ui horizontal segments">
    <Segment
      role="navigation"
      className="breadcrumbs"
      secondary
      vertical
    >
      <Container>
        <Breadcrumb>
          <a href={'/'}>
            Home Page
          </a>
          <Icon name={rightKey} size="18px" />
          <Breadcrumb.Section active>
            e-Learning Courses
          </Breadcrumb.Section>
        </Breadcrumb>
      </Container>
    </Segment>
    <Segment
      role="navigation"
      className="breadcrumbs admin"
      secondary
      vertical
    >
      <Container>
        <Breadcrumb>
          <Breadcrumb.Section>
            admin
          </Breadcrumb.Section>
        </Breadcrumb>
      </Container>
    </Segment>
  </div>
  );

export default Breadcrumbs;
