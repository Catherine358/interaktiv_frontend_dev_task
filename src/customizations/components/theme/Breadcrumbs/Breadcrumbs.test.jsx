import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Breadcrumbs from './Breadcrumbs';

describe('Breadcrumbs', () => {
  it('renders breadcrumbs', async () => {
    const component = render(
      <Breadcrumbs  />
    );
    expect(component).toMatchSnapshot();
    expect(screen.getByText(/home/i)).toBeInTheDocument();
    const link = screen.getByRole('link', { name: 'Home Page'});
    expect(link).toBeInTheDocument();
    expect(link.getAttribute('href')).toBe('/');
  });
});
