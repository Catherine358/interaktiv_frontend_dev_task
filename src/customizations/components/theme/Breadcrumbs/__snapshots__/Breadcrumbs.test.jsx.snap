// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders breadcrumbs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui horizontal segments"
      >
        <div
          class="ui secondary vertical segment breadcrumbs"
          role="navigation"
        >
          <div
            class="ui container"
          >
            <div
              class="ui breadcrumb"
            >
              <a
                href="/"
              >
                Home Page
              </a>
              <svg
                class="icon"
                style="height: 18px; width: auto; fill: currentColor;"
                viewBox=""
                xmlns=""
              />
              <div
                class="active section"
              >
                e-Learning Courses
              </div>
            </div>
          </div>
        </div>
        <div
          class="ui secondary vertical segment breadcrumbs admin"
          role="navigation"
        >
          <div
            class="ui container"
          >
            <div
              class="ui breadcrumb"
            >
              <div
                class="section"
              >
                admin
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui horizontal segments"
    >
      <div
        class="ui secondary vertical segment breadcrumbs"
        role="navigation"
      >
        <div
          class="ui container"
        >
          <div
            class="ui breadcrumb"
          >
            <a
              href="/"
            >
              Home Page
            </a>
            <svg
              class="icon"
              style="height: 18px; width: auto; fill: currentColor;"
              viewBox=""
              xmlns=""
            />
            <div
              class="active section"
            >
              e-Learning Courses
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui secondary vertical segment breadcrumbs admin"
        role="navigation"
      >
        <div
          class="ui container"
        >
          <div
            class="ui breadcrumb"
          >
            <div
              class="section"
            >
              admin
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
